src/main/java/edu/unc/lib/boxc/migration/cdm/services/CdmFieldService.java

```diff
@@ -234,6 +234,85 @@ private void validateFieldName(String field, String headerField, int line, Set<S
         existing.add(field);
     }
 
+    /**
+     * Retrieve field information for the project from the csv
+     * @param
+     * @return
+     */
+    public CdmFieldInfo retrieveFieldsFromCsv(Path exportedObjectsPath) throws IOException {
+        CdmFieldInfo fieldInfo = new CdmFieldInfo();
+
+        try (
+            Reader reader = Files.newBufferedReader(exportedObjectsPath);
+            CSVParser parser = new CSVParser(reader, CSVFormat.DEFAULT
+                    .withTrim());
+        ) {
+            List<String> headers = parser.getRecords().get(0).toList();
+            for (String header : headers) {
+                CdmFieldEntry fieldEntry = new CdmFieldEntry();
+                fieldEntry.setNickName(header);
+                fieldEntry.setExportAs(header);
+                fieldEntry.setDescription(header);
+                fieldEntry.setSkipExport(false);
+                fieldInfo.getFields().add(fieldEntry);
+            }
+        } catch (Exception e) {
+            throw new MigrationException("Failed to parse exported objects path " + exportedObjectsPath, e);
+        }
+        return fieldInfo;
+    }
+
+    /**
+     * Persist the field information out to the project
+     * @param project
+     * @param fieldInfo
+     * @throws IOException
+     */
+    public void persistCsvFieldsToProject(MigrationProject project, CdmFieldInfo fieldInfo) throws IOException {
```

Comment:
I could be wrong, but I feel like we don't need this method and can use `persistFieldsToProject` in place of it? The field info should be populated the same whether it came from CDM or a CSV. I know CDM has way more attributes than we need, but it doesn't do any harm for them to be present and set null/blank.